src/main/java/com/gluonhq/substrate/target/AndroidTargetConfiguration.java

```diff
@@ -380,12 +392,43 @@ private int compileDalvikCode(String androidSrc, String androidJar) throws IOExc
         ProcessRunner processRunner = new ProcessRunner(projectConfiguration.getGraalPath().resolve("bin").resolve("javac").toString(),
                 "-d", getApkClassPath().toString(),
                 "-source", "1.7", "-target", "1.7",
-                "-cp", getApkAndroidSourcePath().toString(),
+                "-cp", getApkAndroidSourcePath().toString() + File.pathSeparator + getApkClassPath().toString(),
                 "-bootclasspath", androidJar);
         processRunner.addArgs(sources);
         return processRunner.runProcess("dalvikCompilation");
     }
 
+    /**
+     * Walks through the jars in the classpath, excluding the JavaFX ones,
+     * and looks for META-INF/substrate/dalvik/*.class files.
+     *
+     * The method will copy all the class files found into the target folder
+     *
+     * @throws IOException
+     */
+    private void copyOtherDalvikClasses() throws IOException, InterruptedException {
+        Path targetFolder = getApkClassPath();
+        Logger.logDebug("Scanning for dalvik classes");
+        final List<File> jars = new ClassPath(projectConfiguration.getClasspath()).getJars(true);
+        String prefix = META_INF_SUBSTRATE_DALVIK + DALVIK_PRECOMPILED_CLASS;
+        for (File jar : jars) {
+            try (ZipFile zip = new ZipFile(jar)) {
+                Logger.logDebug("Scanning " + jar);
+                for (Enumeration e = zip.entries(); e.hasMoreElements(); ) {
+                    ZipEntry zipEntry = (ZipEntry) e.nextElement();
+                    String name = zipEntry.getName();
+                    if (!zipEntry.isDirectory() && name.startsWith(META_INF_SUBSTRATE_DALVIK)) {
+                        Path classPath = targetFolder.resolve(name.substring(prefix.length() + 2));
```

Comment:
Why +2 here?

Reply:
+1 gives "/com/gluonhq/helloandroid...", so resolving "/" goes to root and doesn't take into account `targetFolder`. Adding +2 removes "/" and gives "$targetFolder/com/gluonhq...".

Reply:
I still don't understand. The value of `prefix` is `META-INF/substrate/dalvik/precompiled/class/`. And the name of a zip entry would be `META-INF/substrate/dalvik/precompiled/class/com/hello/MyClass.class`. This means that `name.substring(prefix.length() + 2)` results in the value `m/hello/MyClass.class`?

Reply:
Should be fixed now